.github/workflows/linux_simd.yml

```diff
@@ -67,220 +65,147 @@ jobs:
       with:
         python-version: '3.9'
     - uses: ./.github/meson_actions
+      name: Build/Test
 
   old_gcc:
-    needs: [smoke_test]
-    # provides GCC 8
-    runs-on: ubuntu-20.04
     if: github.event_name != 'push'
-    steps:
-    - uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # v3.6.0
-      with:
-        submodules: recursive
-        fetch-depth: 0
-    - name: Install Python3.9
-      run: |
-        sudo apt update
-        # for add-apt-repository
-        sudo apt install software-properties-common -y
-        sudo add-apt-repository ppa:deadsnakes/ppa -y
-        sudo apt install python3.9-dev ninja-build -y
-        sudo ln -s /usr/bin/python3.9 /usr/bin/pythonx
-        pythonx -m pip install --upgrade pip setuptools wheel
-        pythonx -m pip install -r test_requirements.txt
-    - name: Install GCC 8
-      run: sudo apt install g++-8 -y
-    - name: Build gcc-8
-      run: |
-        export CC=/usr/bin/gcc-8
-        export CXX=/usr/bin/g++-8
-        rm -rf build && pythonx setup.py install --user
-    - name: Run test suite
-      run: pythonx runtests.py -n
-
-  without_optimizations:
-    needs: [smoke_test]
+    needs: [baseline_only]
     runs-on: ubuntu-latest
-    if: github.event_name != 'push'
     env:
-      MESON_ARGS: "-Dallow-noblas=true -Ddisable-optimization=true"
+      MESON_ARGS: "-Dallow-noblas=true"
     steps:
     - uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # v3.6.0
       with:
         submodules: recursive
         fetch-depth: 0
     - uses: actions/setup-python@61a6322f88396a6271a6ee3565807d608ecaddd1 # v4.7.0
       with:
-        python-version: '3.12-dev'
-    - uses: ./.github/meson_actions
+        python-version: '3.9'
+
+    - name: Install GCC/8/9
+      run: |
+        echo "deb http://archive.ubuntu.com/ubuntu focal main universe" | sudo tee /etc/apt/sources.list.d/focal.list
+        sudo apt update
+        sudo apt install -y g++-8 g++-9
+
+    - name: Enable gcc-8
+      run: |
+        sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-8 1
+        sudo update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-8 1
 
-  with_baseline_only:
-    needs: [smoke_test]
-    runs-on: ubuntu-latest
-    if: github.event_name != 'push'
-    env:
-      MESON_ARGS: "-Dallow-noblas=true -Dcpu-dispatch=none"
-    steps:
-    - uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # v3.6.0
-      with:
-        submodules: recursive
-        fetch-depth: 0
-    - uses: actions/setup-python@61a6322f88396a6271a6ee3565807d608ecaddd1 # v4.7.0
-      with:
-        python-version: '3.10'
     - uses: ./.github/meson_actions
+      name: Build/Test against gcc-8
+
+    - name: Enable gcc-9
+      run: |
+        sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-9 1
```

Comment:
This step does not seem to work, the output is only `^@` and afterwards the build log still shows:
```
C compiler for the host machine: cc (gcc 8.4.0 "cc (Ubuntu 8.4.0-3ubuntu2) 8.4.0")
```